src/sql/workbench/services/objectExplorer/browser/objectExplorerViewTreeShim.ts

```diff
@@ -130,19 +130,19 @@ export class OEShimService extends Disposable implements IOEShimService {
 
 	public async getChildren(node: ITreeItem, viewId: string): Promise<ITreeItem[]> {
 		if (node.payload) {
-			let sessionId = await this.getOrCreateSession(viewId, node);
-			let requestHandle = this.nodeHandleMap.get(generateNodeMapKey(viewId, node)) || node.handle;
-			let treeNode = new TreeNode(undefined, undefined, undefined, requestHandle, undefined, undefined, undefined, undefined, undefined, undefined);
+			const sessionId = await this.getOrCreateSession(viewId, node);
+			const requestHandle = this.nodeHandleMap.get(generateNodeMapKey(viewId, node)) || node.handle;
+			const treeNode = new TreeNode(undefined, undefined, undefined, requestHandle, undefined, undefined, undefined, undefined, undefined, undefined);
 			treeNode.connection = new ConnectionProfile(this.capabilities, node.payload);
-			return this.oe.resolveTreeNodeChildren({
+			const childrenNodes = await this.oe.refreshTreeNode({
```

Comment:
I'm assuming this still works even if its the first time the node is being asked for children?

Reply:
Correct. They actually call the same method (refreshOrExpandTreeNode) - but resolveTreeNodeChildren only does a refresh if there's an error message.

https://github.com/microsoft/azuredatastudio/blob/master/src/sql/workbench/services/objectExplorer/browser/objectExplorerService.ts#L577

Reply:
interesting...